ui/components/app/permission-page-container/permission-page-container.component.js

```diff
@@ -54,11 +70,50 @@ export default class PermissionPageContainer extends Component {
 
   getRequestedMethodState(methodNames) {
     return methodNames.reduce((acc, methodName) => {
+      ///: BEGIN:ONLY_INCLUDE_IN(flask)
+      if (methodName === RestrictedMethods.wallet_snap) {
+        acc[methodName] = this.getDedupedSnapPermissions();
+        return acc;
+      }
+      ///: END:ONLY_INCLUDE_IN
       acc[methodName] = true;
       return acc;
     }, {});
   }
 
+  ///: BEGIN:ONLY_INCLUDE_IN(flask)
+  getDedupedSnapPermissions() {
```

Comment:
I'm not sure I understand why we need to do this. Can you explain?

Reply:
Ok, so if a dapp (origin) already has permissions for a snap and is requesting another snap, the second request would also list the first permission the dapp already had and was not requesting, because everything is now under the one permission, `wallet_snap`.

Reply:
i.e. caveat values are merged so the request has all the current snaps + requested snaps, you have to diff to get the actual snaps being requested